includes/admin/class-wc-admin-setup-wizard.php

```diff
@@ -639,46 +706,85 @@ public function wc_setup_shipping() {
 	public function wc_setup_shipping_save() {
 		check_admin_referer( 'wc-setup' );
 
-		if ( isset( $_POST['save_step'] ) && esc_attr( 'Skip this step', 'woocommerce' ) === $_POST['save_step'] ) {
-			update_option( 'woocommerce_ship_to_countries', 'disabled' );
-			wp_redirect( esc_url_raw( $this->get_next_step_link() ) );
-			exit;
-		}
-
-		$current_shipping = get_option( 'woocommerce_ship_to_countries' );
-		$install_services = isset( $_POST['woocommerce_install_services'] );
+		$setup_domestic   = isset( $_POST['shipping_zones']['domestic']['enabled'] ) && ( 'yes' === $_POST['shipping_zones']['domestic']['enabled'] );
+		$domestic_method  = sanitize_text_field( $_POST['shipping_zones']['domestic']['method'] );
+		$setup_intl       = isset( $_POST['shipping_zones']['intl']['enabled'] ) && ( 'yes' === $_POST['shipping_zones']['intl']['enabled'] );
+		$intl_method      = sanitize_text_field( $_POST['shipping_zones']['intl']['method'] );
 		$weight_unit      = sanitize_text_field( $_POST['weight_unit'] );
 		$dimension_unit   = sanitize_text_field( $_POST['dimension_unit'] );
+		$existing_zones   = WC_Shipping_Zones::get_zones();
```

Comment:
It's cleaner to just bail out of the function if `$existing_zones` isn't empty, instead of checking for it in every code block.